dbms/src/TiDB/Schema/TiDB.cpp

```diff
@@ -1362,17 +1398,62 @@ ColumnInfo toTiDBColumnInfo(const tipb::ColumnInfo & tipb_column_info)
     tidb_column_info.flag = tipb_column_info.flag();
     tidb_column_info.flen = tipb_column_info.columnlen();
     tidb_column_info.decimal = tipb_column_info.decimal();
+    tidb_column_info.collate = tipb_column_info.collation();
+    for (int i = 0; i < tipb_column_info.elems_size(); ++i)
+        tidb_column_info.elems.emplace_back(tipb_column_info.elems(i), i + 1);
     // TiFlash get default value from origin_default_value, check `Field ColumnInfo::defaultValueToField() const`
     // So we need to set origin_default_value to tipb_column_info.default_val()
     // Related logic in tidb, https://github.com/pingcap/tidb/blob/45318da24d8e4c0c6aab836d291a33f949dd18bf/pkg/table/tables/tables.go#L2303-L2329
-    // For TypeBit, we need to set origin_default_bit_value to tipb_column_info.default_val().
-    if (tidb_column_info.tp == TypeBit)
-        tidb_column_info.origin_default_bit_value = tipb_column_info.default_val();
-    else
+    if (tidb_column_info.tp != TypeBit)
         tidb_column_info.origin_default_value = tipb_column_info.default_val();
-    tidb_column_info.collate = tipb_column_info.collation();
-    for (int i = 0; i < tipb_column_info.elems_size(); ++i)
-        tidb_column_info.elems.emplace_back(tipb_column_info.elems(i), i + 1);
+    // Decode tipb_column_info.default_val.
+    {
+        // The default value is null.
+        if (tidb_column_info.origin_default_value.size() == 0)
```

Comment:
```suggestion
        if (tidb_column_info.origin_default_value.isEmpty())
```

Reply:
I remember I tried this. These two are not the same value. Just make sure.

Reply:
ok